packages/data/src/world/mm/great_bay_temple.yml

```diff
@@ -180,42 +180,42 @@
   events:
-    GB_PIPE_GREEN2: "(can_use_ice_arrows && (can_use_fire_arrows || trick(MM_GBT_FIRELESS))) || short_hook_anywhere"
+    GB_PIPE_GREEN2: "(can_use_ice_arrows && (can_use_fire_arrows || trick(MM_GBT_FIRELESS))) || short_hook_anywhere || has_hover_boots"
   locations:
-    "Great Bay Temple Green Pipe 3 Chest": "(can_use_ice_arrows && (can_use_fire_arrows || trick(MM_GBT_FIRELESS)) && can_hookshot) || short_hook_anywhere"
-    "Great Bay Temple SF Green Pipe 3 Barrel": "has_mask_zora || ((has_bombchu || has(MASK_BLAST)) && short_hook_anywhere)"
-    "Great Bay Temple Pot Green Pipe 3 Upper 1": "(can_use_ice_arrows && (can_use_fire_arrows || trick(MM_GBT_FIRELESS))) || short_hook_anywhere"
-    "Great Bay Temple Pot Green Pipe 3 Upper 2": "(can_use_ice_arrows && (can_use_fire_arrows || trick(MM_GBT_FIRELESS))) || short_hook_anywhere"
+    "Great Bay Temple Green Pipe 3 Chest": "(can_use_ice_arrows && (can_use_fire_arrows || trick(MM_GBT_FIRELESS)) && can_hookshot) || short_hook_anywhere || has_hover_boots"
+    "Great Bay Temple SF Green Pipe 3 Barrel": "underwater_walking || ((has_bombchu || has(MASK_BLAST)) && short_hook_anywhere)"
+    "Great Bay Temple Pot Green Pipe 3 Upper 1": "(can_use_ice_arrows && (can_use_fire_arrows || trick(MM_GBT_FIRELESS))) || short_hook_anywhere || has_hover_boots"
+    "Great Bay Temple Pot Green Pipe 3 Upper 2": "(can_use_ice_arrows && (can_use_fire_arrows || trick(MM_GBT_FIRELESS))) || short_hook_anywhere || has_hover_boots"
     "Great Bay Temple Pot Green Pipe 3 Lower": "true"
 "Great Bay Temple Pre-Boss":
   dungeon: GB
   exits:
     "Great Bay Temple Boss Access": "boss_key(BOSS_KEY_GB) && ((event(GB_PIPE_GREEN) && event(GB_PIPE_GREEN2)) || short_hook_anywhere)"
   locations:
     "Great Bay Temple SF Pre-Boss Above Water": "can_use_ice_arrows || (has(MASK_GREAT_FAIRY) && (has_arrows || can_hookshot)) || short_hook_anywhere"
-    "Great Bay Temple SF Pre-Boss Underwater": "has_mask_zora || (can_use_ice_arrows && (has(MASK_GREAT_FAIRY) || short_hook_anywhere))"
-    "Great Bay Temple Pot Pre-Boss 1": "has_mask_zora || hookshot_anywhere"
-    "Great Bay Temple Pot Pre-Boss 2": "has_mask_zora || hookshot_anywhere"
-    "Great Bay Temple Pot Pre-Boss 3": "has_mask_zora || hookshot_anywhere"
-    "Great Bay Temple Pot Pre-Boss 4": "has_mask_zora || hookshot_anywhere"
-    "Great Bay Temple Pot Pre-Boss 5": "has_mask_zora || hookshot_anywhere"
-    "Great Bay Temple Pot Pre-Boss 6": "has_mask_zora || hookshot_anywhere"
-    "Great Bay Temple Pot Pre-Boss 7": "has_mask_zora || hookshot_anywhere"
-    "Great Bay Temple Pot Pre-Boss 8": "has_mask_zora || hookshot_anywhere"
+    "Great Bay Temple SF Pre-Boss Underwater": "has_mask_zora || (can_use_ice_arrows && (has(MASK_GREAT_FAIRY) || short_hook_anywhere || underwater_walking)) || (underwater_walking && can_hookshot_short)"
```

Comment:
Zora can just swim into the bubble and grab the fairy. I would imagine Irons can do the same. The short hook anywhere method also just goes through the bubble.

Reply:
Zora pops the bubble with fins then swims into it, doesn't it? If you pop the bubble with your body you'll go past it and when you come back the room the fairy is back in the bubble. The ice arrow methods can pop the bubble with an arrow before swimming into it. So imo this doesnt need to change.